report/report.go

```diff
@@ -165,40 +165,39 @@ func ReportRuntime(_ monitoring.Mode, V monitoring.Visitor) {
 }
 
 func InstanceCroupsReporter(logger *logp.Logger, override string) func(monitoring.Mode, monitoring.Visitor) {
+	// PID shouldn't use hostfs, at least for now.
+	// containerization schemes should provide their own /proc/ that will serve containerized processess
+	pid := os.Getpid()
+
+	cgroups, err := cgroup.NewReaderOptions(cgroup.ReaderOptions{
+		RootfsMountpoint:         resolve.NewTestResolver("/"),
+		IgnoreRootCgroups:        true,
+		CgroupsHierarchyOverride: os.Getenv(override),
+	})
+	if err != nil {
+		if errors.Is(err, cgroup.ErrCgroupsMissing) {
+			logger.Warnf("cgroup data collection disabled in internal monitoring: %v", err)
```

Comment:
This change also changes how warnings and errors are reported: instead of on every callback function call, only once before the callback is ever executed. I suppose it's not a bad change, perhaps a good one. Just wanted to make sure it's noted.